core/storageproviders/local.go

```diff
@@ -42,8 +45,16 @@ func (s *LocalStorage) VariantPlaylistWritten(localFilePath string) {
 
 // MasterPlaylistWritten is called when the master hls playlist is written.
 func (s *LocalStorage) MasterPlaylistWritten(localFilePath string) {
-	if _, err := s.Save(localFilePath, 0); err != nil {
-		log.Warnln(err)
+
+	// If we're using a remote serving endpoint, we need to rewrite the master playlist
+	if s.host != "" {
+		if err := rewriteRemotePlaylist(localFilePath, s.host, ""); err != nil {
```

Comment:
Originally, I called this `rewriteRemotePlaylist` because it was only used for rewriting a local playlist to be used as a remote playlist. I feel like the name is no longer accurate. Maybe it should be named to `rewritePlaylistLocations` or something? Can you think of a more representative name?

Reply:
True, it might be more accurate to give it a more generic name
Any particular reason you used the plural `rewritePlaylistLocations` instead of `rewritePlaylistLocation`?

Reply:
The master playlist holds multiple media playlists, each has its own URL. So it needs to rewrite each one.